clu/src/lib.rs

```diff
@@ -130,6 +133,45 @@ fn linux_init(config: Arc<RwLock<settings::RitaSettingsStruct>>) -> Result<(), E
         },
     }
 
+    match device_option {
+        Some(existing_device) => {
+            info!("Device name is {}", existing_device);
+        }
+        None => {
+            let release_file_path = "/etc/althea-firmware-release";
+            info!(
+                "No device name was found, reading from {}",
+                release_file_path
+            );
+
+            let mut contents = String::new();
+            match File::open(release_file_path) {
+                Ok(mut f) => {
+                    f.read_to_string(&mut contents)?;
+                }
+                Err(e) => warn!("Couldn't open {}: {}", release_file_path, e),
+            };
+
+            for line in contents.lines() {
+                if line.starts_with("device:") {
+                    let mut array = line.split(" ");
+                    array.next();
+                    let device = array.next();
```

Comment:
Looks like you just want the second element, in that case just go `let device = line.split(" ").nth(2).ok_or(format_err!("Could not obtain device name from line {:?}", line))?`. Here's what it does:
* `nth(arg)` does `next()` `arg+1` times and gives you the `Option` from the last `next()`
* `ok_or(arg)` converts the `Option` from `nth` to a `Result`. It'll return `Ok(val)` if the `Option` was `Some(whatever)` and `Err(arg)` otherwise. The `arg` is `ok_or`'s argument.
* `format_err!()` macro is a helper that'll build an error type from whatever formatted string you supply it with.
*  `?` after an expression does `return Err(something)` if the result of the expression was `Err(something)` and if it was an `Ok(something_else` it just unwraps it into just `something_else`. Analogically with `Option`'s but with `Some` and `None` in place of `Ok` and `Err`, respectively.